packages/shorebird_cli/lib/src/auth/auth.dart

```diff
@@ -320,10 +330,10 @@ class UserNotFoundException implements Exception {
 
 extension OauthAuthProvider on Jwt {
   oauth2.AuthProvider get authProvider {
-    if (payload.iss.startsWith('https://login.microsoftonline.com')) {
-      return MicrosoftAuthProvider();
-    } else if (payload.iss == 'https://accounts.google.com') {
+    if (googleJwtIssuerRegexp.hasMatch(payload.iss)) {
```

Comment:
Instead of having this thing do the iss -> auth provider, you might consider having a middle step which is iss string -> enum -> auth-provider.  Then the string -> enum/object could hide all the regexp goop.  🤷